### ISSUE TITLE
CI: Mac

### ISSUE THREAD

**Issue by angerman** [snowleopard/shaking-up-ghc#111]
Travis allows to run on their os x infrastructure as well. See https://docs.travis-ci.com/user/workers/os-x-infrastructure/

**Comment by snowleopard** [comment]
I'll reopen this: see https://travis-ci.org/snowleopard/shaking-up-ghc/jobs/100553205. It seems that we don't have `clang` on travis:
```
Error when running Shake build system:
* inplace/bin/hp2ps
* utils/hp2ps/stage0/build/Utilities.o
* OracleQ (DependenciesKey ("utils/hp2ps/stage0/build/.dependencies","utils/hp2ps/stage0/build/Utilities.o"))
* utils/hp2ps/stage0/build/.dependencies
* utils/hp2ps/stage0/build/Utilities.c.deps
* clang
Error, file does not exist and no rule available:
  clang
```

**Comment by angerman** [comment]
>  It seems that we don't have clang on travis:
> 
They do. I'll check when I'm back home.

**Comment by snowleopard** [comment]
Ah, I think I know the reason. We are missing `./configure --with-gcc=$(which clang)`.

**Comment by angerman** [comment]
> Ah, I think I know the reason. We are missing ./configure --with-gcc=$(which clang).
> 
You are right. That could be the reason. On second consideration, I might have an idea how to fix that `clang` issue :-)
>

**Comment by snowleopard** [comment]
After these two fixes we have a working Mac CI: 03b3379, b67f727.

The latter one reduces the build time beyond 50 min (hard limit for OS X) by requiring `base` library as the main target.
